version-compatibility/src/fuel_tx.rs

```diff
@@ -26,6 +26,35 @@ fn release_0_58_2_can_deserialize_latest() {
     let _ = latest_tx.expect("Deserialization failed");
 }
 
+#[test]
+fn latest_can_deserialize_previous_tx_pointer() {
+    for idx in 0..=u16::MAX {
+        // Given
+        let tx_pointer = fuel_tx_0_59_1::TxPointer::new(1u32.into(), idx);
+        let expected = latest_fuel_tx::TxPointer::new(1u32.into(), idx.into());
```

Comment:
May I ask you to do the same for script transaction please?=D Or `Mint` transaction for simplicity

Reply:
I tried it requires a lot of machinery to specify a `TxPointer` inside the transaction. At the top of the file there is the test `latest_can_deserialize_0_58_2` that do this with a test tx. `TxPointer` is set to 0 but if the deserialization wasn't working then it would flag as a error for the number of bytes. Why do you think a test for this is still valuable ?

Reply:
Because based on the value of `idx` the result serialization can be different. It why test with `0` tx pointer will not catch the problem

Reply:
Found a way to do it without too much machinery. I hope it's fine for you :)